tests/unit/TestNetRemoteDataStreamingServiceClient.cxx

```diff
@@ -71,3 +72,48 @@ TEST_CASE("DataStreamUpload API", "[basic][rpc][client][remote][stream]")
         }
     }
 }
+
+TEST_CASE("DataStreamDownload API", "[basic][rpc][client][remote][stream]")
+{
+    using namespace Microsoft::Net::Remote;
+    using namespace Microsoft::Net::Remote::DataStream;
+    using namespace Microsoft::Net::Remote::Service;
+
+    using Microsoft::Net::Remote::Test::DataStreamReader;
+    using Microsoft::Net::Remote::Test::RemoteServiceAddressHttp;
+
+    NetRemoteServerConfiguration Configuration{
+        .ServerAddress = RemoteServiceAddressHttp,
+    };
+
+    NetRemoteServer server{ Configuration };
+    server.Run();
+
+    auto channel = grpc::CreateChannel(RemoteServiceAddressHttp, grpc::InsecureChannelCredentials());
+    auto client = NetRemoteDataStreaming::NewStub(channel);
+
+    static constexpr auto fixedNumberOfDataBlocksToStream = 10;
+
+    SECTION("Can be called with DataStreamTypeFixed and DataStreamPatternConstant")
+    {
+        DataStreamFixedTypeProperties fixedTypeProperties{};
+        fixedTypeProperties.set_numberofdatablockstostream(fixedNumberOfDataBlocksToStream);
+
+        DataStreamProperties properties{};
+        properties.set_type(DataStreamType::DataStreamTypeFixed);
+        properties.set_pattern(DataStreamPattern::DataStreamPatternConstant);
+        *properties.mutable_fixedtypeproperties() = std::move(fixedTypeProperties);
+
+        DataStreamDownloadRequest request{};
+        *request.mutable_properties() = std::move(properties);
+
+        auto dataStreamReader = std::make_unique<DataStreamReader>(client.get(), &request);
```

Comment:
Any reason this can't be allocated on the stack instead? Eg.

```C++
DataStreamReader dataStreamReader{  client.get(), &request };
// ...
grpc::Status status = dataStreamReader.Await(&numberOfDataBlocksReceived, &operationStatus);
```
```

Reply:
I believe it can be stack allocated. Will update